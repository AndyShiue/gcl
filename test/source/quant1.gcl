b := <| (+) i : 0 <= i < k : F i |>
n

{ true }
