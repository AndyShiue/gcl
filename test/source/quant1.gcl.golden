Right (Program [] [] [] [Assign [Name "b" (Loc (Pos "./test/source/quant1.gcl" 1 1 0) (Pos "./test/source/quant1.gcl" 1 1 0))] [App (Quant (Op Add (Loc (Pos "./test/source/quant1.gcl" 1 9 8) (Pos "./test/source/quant1.gcl" 1 11 10))) [Name "i" (Loc (Pos "./test/source/quant1.gcl" 1 13 12) (Pos "./test/source/quant1.gcl" 1 13 12))] (App (App (Op LT (Loc (Pos "./test/source/quant1.gcl" 1 24 23) (Pos "./test/source/quant1.gcl" 1 24 23))) (App (App (Op LTE (Loc (Pos "./test/source/quant1.gcl" 1 19 18) (Pos "./test/source/quant1.gcl" 1 20 19))) (Lit (Num 0) (Loc (Pos "./test/source/quant1.gcl" 1 17 16) (Pos "./test/source/quant1.gcl" 1 17 16))) (Loc (Pos "./test/source/quant1.gcl" 1 17 16) (Pos "./test/source/quant1.gcl" 1 20 19))) (Var (Name "i" (Loc (Pos "./test/source/quant1.gcl" 1 22 21) (Pos "./test/source/quant1.gcl" 1 22 21))) (Loc (Pos "./test/source/quant1.gcl" 1 22 21) (Pos "./test/source/quant1.gcl" 1 22 21))) (Loc (Pos "./test/source/quant1.gcl" 1 17 16) (Pos "./test/source/quant1.gcl" 1 22 21))) (Loc (Pos "./test/source/quant1.gcl" 1 17 16) (Pos "./test/source/quant1.gcl" 1 24 23))) (Var (Name "k" (Loc (Pos "./test/source/quant1.gcl" 1 26 25) (Pos "./test/source/quant1.gcl" 1 26 25))) (Loc (Pos "./test/source/quant1.gcl" 1 26 25) (Pos "./test/source/quant1.gcl" 1 26 25))) (Loc (Pos "./test/source/quant1.gcl" 1 17 16) (Pos "./test/source/quant1.gcl" 1 26 25))) (App (Const (Name "F" (Loc (Pos "./test/source/quant1.gcl" 1 30 29) (Pos "./test/source/quant1.gcl" 1 30 29))) (Loc (Pos "./test/source/quant1.gcl" 1 30 29) (Pos "./test/source/quant1.gcl" 1 30 29))) (Var (Name "i" (Loc (Pos "./test/source/quant1.gcl" 1 32 31) (Pos "./test/source/quant1.gcl" 1 32 31))) (Loc (Pos "./test/source/quant1.gcl" 1 32 31) (Pos "./test/source/quant1.gcl" 1 32 31))) (Loc (Pos "./test/source/quant1.gcl" 1 30 29) (Pos "./test/source/quant1.gcl" 1 32 31))) (Loc (Pos "./test/source/quant1.gcl" 1 6 5) (Pos "./test/source/quant1.gcl" 1 35 34))) (Var (Name "n" (Loc (Pos "./test/source/quant1.gcl" 2 1 36) (Pos "./test/source/quant1.gcl" 2 1 36))) (Loc (Pos "./test/source/quant1.gcl" 2 1 36) (Pos "./test/source/quant1.gcl" 2 1 36))) (Loc (Pos "./test/source/quant1.gcl" 1 6 5) (Pos "./test/source/quant1.gcl" 2 1 36))] (Loc (Pos "./test/source/quant1.gcl" 1 1 0) (Pos "./test/source/quant1.gcl" 2 1 36)),Assert (Var (Name "true" (Loc (Pos "./test/source/quant1.gcl" 4 3 41) (Pos "./test/source/quant1.gcl" 4 6 44))) (Loc (Pos "./test/source/quant1.gcl" 4 3 41) (Pos "./test/source/quant1.gcl" 4 6 44))) (Loc (Pos "./test/source/quant1.gcl" 4 1 39) (Pos "./test/source/quant1.gcl" 4 8 46))] (Loc (Pos "./test/source/quant1.gcl" 1 1 0) (Pos "./test/source/quant1.gcl" 4 8 46)))